analyzer/src/main/kotlin/managers/Pip.kt

```diff
@@ -381,14 +381,14 @@ class Pip(
         )
     }
 
-    private fun getBinaryArtifact(pkg: Package, releaseNode: ArrayNode): RemoteArtifact {
+    private fun getBinaryArtifact(releaseNode: ArrayNode): RemoteArtifact {
         // Prefer python wheels and fall back to the first entry (probably a sdist).
         val binaryArtifact = releaseNode.asSequence().find {
             it["packagetype"].textValue() == "bdist_wheel"
         } ?: releaseNode[0]
 
-        val url = binaryArtifact["url"]?.textValue() ?: pkg.binaryArtifact.url
-        val hash = binaryArtifact["md5_digest"]?.textValue()?.let { Hash.create(it) } ?: pkg.binaryArtifact.hash
+        val url = binaryArtifact["url"]?.textValue() ?: return RemoteArtifact.EMPTY
+        val hash = binaryArtifact["md5_digest"]?.textValue()?.let { Hash.create(it) } ?: return RemoteArtifact.EMPTY
```

Comment:
I don't think the values of `url` and `hash` were always empty at this point, so we are probably losing data with this change.

Reply:
Hm, I thought they were always empty, because of the constructor call of the `Package` in `Pip.kt`.
Why do you think they were not?

Reply:
I thought that because they were coming from the `packageTemplates` variable, but it turns out in `parseDependencies` no binary artifact is set. It was probably implemented like that in case other sources for artifacts would be added later.